streaming/base/util.py

```diff
@@ -209,6 +215,120 @@ def get_import_exception_message(package_name: str, extra_deps: str) -> str:
             f'`pip install \'mosaicml-streaming[{package_name}]\'`.'
 
 
+def do_merge_index(index_file_urls: List[Any],
+                   out: Union[str, Tuple[str, str]],
+                   keep_local: bool = True,
+                   download_timeout: int = 60) -> None:
+    """Merge index.json from a list of index.json. Write to `out`, overwriting if exists.
```

Comment:
Can you please add more details ? something like: `Merge index.json from a list of subset of dataset's index.json to create a one joined dataset`. You can come up with better wordings lol